templates/samples.html

```diff
@@ -0,0 +1,229 @@
+
+var t = $('#t');
+t.categories = {{categories|safe}}; // server included.
+t.prevSelectedCategory = null;
+
+t._computeIcon = function(el) {
+  return 'chromestatus:' + (el ? el.dataset.category : '');
+};
+
+t._computeIconId = function(categoryStr) {
+  return 'chromestatus:' + this.categories[categoryStr];
+};
+
+t._computeFeatureLink = function(id) {
+  return '/features/' + id;
+};
+
+t._onSelectCategory = function(e, detail) {
+  // Enable toggling selected item until
+  // https://github.com/PolymerElements/paper-menu/issues/50 is fixed.
+  if (this.selectedCategory === this.prevSelectedCategory) {
+    this.$.papermenu.selected = null;
+    this.prevSelectedCategory = null;
+  } else {
+    this.prevSelectedCategory = this.selectedCategory;
+  }
+
+  this.filter(search.value);
+};
+
+t._onRippleDone = function(e, detail) {
+  // Needed until https://github.com/PolymerElements/paper-ripple/pull/41 is fixed.
+  this.debounce('ripple', function() {
+    t._onSelectCategory();
+  }, 10);
+};
+
+t.filter = function(val) {
+  var features = this.features;
+  if (val) {
+    var regex = new RegExp(val, 'i');
+    features = features.filter(function(feature, idx, array) {
+      return regex.test(feature.name) || regex.test(feature.summary);
+    });
+  }
+
+  // Further refine list based on selected category in menu.
+  if (this.selectedCategory) {
+    var category = this.selectedCategory.textContent.trim();
+    var regex = new RegExp(category, 'i');
+    features = features.filter(function(feature, idx, array) {
+      return regex.test(feature.category);
+    });
+  }
+
+  this.filtered = features;
+};
+
+t.removeFilters = function() {
+  t.filter();
+};
+
+// Fire of samples.json XHR right away so data can populate faster.
+var samplesPromise = new Promise(function(resolve, reject) {
+  var url = location.hostname == 'localhost' ? 'https://polymer10-samples-dot-cr-status.appspot.com/samples.json' :
+                                               '/samples.json';
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  xhr.responseType = 'json';
+  xhr.onloadend = function(e) {
+    if (e.target.status < 400) {
+      resolve(e.target.response);
+    } else {
+      reject(new Error('Samples XHR failed with status ' + e.target.statusText));
+    }
+  };
+  xhr.onerror = reject;
+  xhr.send();
+});
+
+samplesPromise.then(function(features) {
+  var re = /github.com\/GoogleChrome\/samples\/tree\/gh-pages\/(.*)/;
+  features.forEach(function(f, i) {
+    f.sample_links.forEach(function(link, i) {
+      var match = re.test(link);
+      if (match) {
+        f.sample_links[i] = link.replace(re, 'googlechrome.github.io/samples/$1');
+      }
+    });
+  });
+
+  t.features = features;
+  t.filtered = features;
+
+  $('#sample-panel').classList.remove('loading');
+
+}, function(e) {
+  console.error(e);
+});
+
+// Clear input when user clicks the 'x' button.
+search.addEventListener('search', function(e) {
```

Comment:
Would you believe `input[type="search"` fires this!??!?! It's the event when someone clicks the "x"/